src/utils.ts

```diff
@@ -0,0 +1,138 @@
+import { PartialFileList } from './PartialFile';
+
+export type StringObject = Record<string, string>;
+
+/**
+ * Type for [[`groupFiles`]] function
+ */
+export type GroupFiles = (
+  flist: PartialFileList | FileList,
+  options: GroupFilesOptions,
+) => GroupedFiles[];
+/**
+ * Group files under same directory path
+ * i.e `[Dir1, Dir2, Dir3...]`  where each Dir is
+ * ```
+ * {
+ *   id,
+ *   extension,
+ *   name,
+ *   captureGroups
+ *   fileA,
+ *   fileB,...
+ *  }
+ *  ```
+ * Files must be in the same directory to be grouped.
+ * @param fileList
+ * @param options - how to build the out object. See `[[GroupFilesOptions]]` for all options.
+ * @return - array of dirs. Some keys are File objects (`fileA`, `fileB`..).
+ */
+
+export const groupFiles: GroupFiles = (
```

Comment:
We like to have as filename the same name as the function we export. Please rename utils and utils.test to groupFiles and groupFiles.test

Reply:
We should enver declare a function this way because it yields to an anonnymous function that can not be debugged easily later. I'm surprised that it is even allowed by our rules.

Please use something like:

`export function groupFiles(fileList, options): GroupFiles`